src/Servers/Kestrel/Core/src/Internal/Http2/Http2Stream.cs

```diff
@@ -71,7 +71,10 @@ public void Initialize(Http2StreamContext context)
         public void InitializeWithExistingContext(int streamId)
         {
             _context.StreamId = streamId;
-            Initialize(_context);
+
+            // The stream was reset when it was completed.
+            // We don't want to reset it twice because reused headers will be discarded.
+            Initialize(_context, reset: false);
```

Comment:
The idea is if we are using a pooled stream, we don't want to call reset again, right. 

Let's rename this method to make it more clear that it should only be called for pooled stream.

Reply:
I'm not sure what you want. Do you want it renamed to InitializePooled? This is the method on HttpProtocol. Do you want all the initialize methods on Http1Connection, Http2Stream and Http3Stream renamed as well? Is that adding much information? HttpProtocol is always pooled.

What information do you want to give to developers who are reading this code?

Reply:
I've renamed the Initialize method on Http2Stream and added some XML docs to it.